controllers/etcd_controller.go

```diff
@@ -1082,28 +1085,12 @@ func (r *EtcdReconciler) updateEtcdErrorStatus(etcd *druidv1alpha1.Etcd, sts *ap
 }
 
 func (r *EtcdReconciler) updateEtcdStatus(etcd *druidv1alpha1.Etcd, svc *corev1.Service, sts *appsv1.StatefulSet) error {
-
-	svcName := svc.Name
-	etcd.Status.Etcd = druidv1alpha1.CrossVersionObjectReference{
-		APIVersion: sts.APIVersion,
-		Kind:       sts.Kind,
-		Name:       sts.Name,
-	}
 	ready := health.CheckStatefulSet(sts) == nil
-	conditions := []druidv1alpha1.Condition{}
-	for _, condition := range sts.Status.Conditions {
-		conditions = append(conditions, convertConditionsToEtcd(&condition))
-	}
-	etcd.Status.Conditions = conditions
-
-	// To be changed once we have multiple replicas.
-	etcd.Status.CurrentReplicas = sts.Status.CurrentReplicas
-	etcd.Status.ReadyReplicas = sts.Status.ReadyReplicas
-	etcd.Status.UpdatedReplicas = sts.Status.UpdatedReplicas
+	etcd.Status.Ready = &ready
```

Comment:
In principle, we could move this to the custodian too 🙂

Reply:
It is there in custodian as well. But if we completely remove this from here, etcd_controller is not getting reconciled properly.. throwing error in testing

Reply:
👍 We can clean this up separately while refactoring the tests.